docs/rules/v-on-handler-style.md

```diff
@@ -0,0 +1,153 @@
+
+> enforce writing style for handlers in `v-on` directives
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+This rule aims to enforce using method handlers on `v-on`, using inline handlers on `v-on`, or binding inline functions to `v-on`.
+
+```vue
+<!-- Method handlers -->
+<button v-on:click="handler">...</button>
+<!-- Inline handlers -->
+<button v-on:click="handler()">...</button>
+<!-- Inline functions -->
+<button v-on:click="() => handler()">...</button>
+```
+
+<eslint-code-block fix :rules="{'vue/v-on-handler-style': ['error']}">
+
+```vue
+<template>
+  <!-- ✓ GOOD -->
+  <button v-on:click="handler">...</button>
+  <button v-on:click="() => handler()">...</button>
+
+  <!-- ✗ BAD -->
+  <button v-on:click="handler()">...</button>
+</template>
+```
+
+</eslint-code-block>
+
+## :wrench: Options
+
+```json
+{
+  "vue/v-on-handler-style": ["error",
+    [
+      "method",
+      "inline-function"
+    ],
+    {
+      "ignoreIncludesComment": false
+    }
+  ]
+}
+```
+
+- First option ... An array of names of allowed styles. Default is `["method", "inline-function"]`.
+  - `"method"` ... Allow handlers by method binding. e.g. `v-on:click="handler"`
+  - `"inline"` ... Allow inline handlers. e.g. `v-on:click="handler()"`  
+                  Even if this is not specified, writing styles that cannot be converted to other allowed styles are allowed.
+  - `"inline-function"` ... Allow inline functions. e.g. `v-on:click="() => handler()"`  
+                            Even if this is not specified, writing styles that cannot be converted to other allowed styles are allowed.
+- Second option
+  - `ignoreIncludesComment` ... If `true`, do not report expressions containing comments. This option takes effect if `"method"` is allowed. Default is `false`.
+
+### `["method"]`
+
+<eslint-code-block fix :rules="{'vue/v-on-handler-style': ['error', ['method']]}">
+
+```vue
+<template>
+  <!-- ✓ GOOD -->
+  <button v-on:click="handler">...</button>
+
+  <!-- ✗ BAD -->
+  <button v-on:click="handler()">...</button>
+  <button v-on:click="() => handler()">...</button>
+
+  <!-- Ignore -->
+  <button v-on:click="handler(foo)">...</button>
+  <button v-on:click="() => handler(foo)">...</button>
+</template>
+```
+
+</eslint-code-block>
+
+### `["inline"]`
+
+<eslint-code-block fix :rules="{'vue/v-on-handler-style': ['error', ['inline']]}">
+
+```vue
+<template>
+  <!-- ✓ GOOD -->
+  <button v-on:click="handler()">...</button>
+
+  <!-- ✗ BAD -->
+  <button v-on:click="handler">...</button>
+  <button v-on:click="() => handler()">...</button>
+  <button v-on:click="(foo) => handler(foo)">...</button>
+</template>
```

Comment:
Please also add an example and test for ignored style here (and check if it is actually ignored):

```vue
<template>
  <!-- ✓ GOOD -->
  <button v-on:click="handler()">...</button>

  <!-- ✗ BAD -->
  <button v-on:click="handler">...</button>
  <button v-on:click="() => handler()">...</button>
  <button v-on:click="(foo) => handler(foo)">...</button>

  <!-- Ignore -->
  <button v-on:click="(foo, bar) => handler(foo, bar)">...</button>
  <button v-on:click="handler">...</button> <!-- if handler has multiple parameters and/or event has multiple payload parameters; don't know if we can check that -->
</template>
```

Reply:
Or, on a second thought: Maybe we should rather simplify the a lot logic by not ignoring anything? If the user wants a specific pattern, better enable it in the rule? By default, everything can be achieved by either the method or the inline style, so the default is fine anyway.